metriq_gym/benchmarks/quantum_volume.py

```diff
@@ -1,45 +1,219 @@
+        confidence_level=confidence_level,
+        confidence_pass=p_val < confidence_level,
+        eplg=(1 - (xeb ** (1 / n))) if xeb < 1 else 0,
+    )
+
+
+def calc_stats(data: QuantumVolumeData, counts: list[MeasCount]) -> AggregateStats:
+    """Calculate aggregate statistics over multiple trials.
+
+    Args:
+        data: contains dispatch-time data (input data + ideal probability).
+        counts: contains results from the quantum device (one MeasCount per trial).
+    Returns:
+        A list of `AggregateStats` objects, each containing aggregated statistics for a result.
+    """
+    trial_stats = []
+
+    num_trials = len(counts)
+    # Process each trial, handling provider-specific logic.
+    for trial in range(num_trials):
+        trial_stat = calc_trial_stats(
+            ideal_probs=data.ideal_probs[trial],
+            counts=counts[trial],
+            shots=data.shots,
+            confidence_level=data.confidence_level,
         )
+        trial_stats.append(trial_stat)
+
+    # Aggregate the trial statistics.
+    hog_prob = sum(stat.hog_prob for stat in trial_stats) / num_trials
+    p_value = math.prod(stat.p_value for stat in trial_stats) ** (1 / num_trials)
 
-        if len(result.counts) > 0:
-            stats = calc_stats([result])
-            logging.info(f"Simulator-only job completed: {stats[0]}.")
-            print(stats[0])
-            return
+    return AggregateStats(
+        trials=num_trials,
+        trial_p_values=[stat.p_value for stat in trial_stats],
+        hog_prob=hog_prob,
+        p_value=p_value,
+        hog_pass=all(stat.hog_pass for stat in trial_stats),
+        confidence_pass=all(stat.confidence_pass for stat in trial_stats),
+    )
 
-        logging.info(f"Dispatched {self.params["trials"]} trials in 1 job.")
 
-        self.job_manager.add_job(result.to_serializable())
+class QuantumVolume(Benchmark):
+    def dispatch_handler(self, device: QuantumDevice) -> QuantumVolumeData:
+        num_qubits = self.params.num_qubits
+        shots = self.params.shots
+        trials = self.params.trials
+        circuits, ideal_probs = prepare_qv_circuits(num_qubits, trials)
+        quantum_job: QuantumJob | list[QuantumJob] = device.run(circuits, shots=shots)
+        job_data = QuantumVolumeData(
+            provider_job_id=[quantum_job.id]
+            if isinstance(quantum_job, QuantumJob)
+            else [job.id for job in quantum_job],
+            qubits=num_qubits,
+            shots=shots,
+            depth=num_qubits,
+            confidence_level=self.params.confidence_level,
+            ideal_probs=ideal_probs,
+            trials=trials,
+        )
+        return job_data
 
-        logging.info("Dispatched job saved.")
+    def poll_handler(self, job_data: BenchmarkData, result_data: list[ResultData]) -> None:
+        if not isinstance(job_data, QuantumVolumeData):
+            raise TypeError("Expected job_data to be of type QuantumVolumeJobData")
```

Comment:
```suggestion
            raise TypeError(f"Expected job_data to be of type {type(QuantumVolumeJobData)}")
```

The reason for this suggestion is if we change the name of this, we make sure that the message isn't out of sync with the actual name of the data object.